server/utils/secrets/secret_ref_utils.py

```
@@ -80,16 +81,37 @@ def _resolve_org(user_id: str) -> Optional[str]:
 
-    Returns (sql_fragment, params) to append to a WHERE clause.
-    When org_id is available the fragment restricts to matching rows;
-    otherwise it returns an always-true fragment so the query still works.
+    This sanitizes user-supplied IDs so taint-analysis tools can confirm
+    that only structurally valid values reach database queries.
+    Raises ValueError if the format is invalid.
     """
+    try:
+        _uuid.UUID(str(value))
+    except (ValueError, AttributeError):
+        raise ValueError(f"Invalid {label} format: {value!r}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)